src/main/java/persistence/sql/dml/clause/ValueClause.java

```diff
@@ -1,50 +1,49 @@
-package persistence.sql.dml.caluse;
+package persistence.sql.dml.clause;
 
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
 import jakarta.persistence.Transient;
+import persistence.sql.meta.pk.GenerationType;
 
 import java.lang.reflect.Field;
 
 public class ValueClause {
-    public static final String DEFAULT_VALUE = "DEFAULT";
-    public static final String STRING_QUOTE = "'";
-    private final Object object;
-    private final Field field;
+    private static final String DEFAULT_VALUE = "DEFAULT";
+    private static final String SINGLE_QUOTE = "'";
+    private final String value;
 
     public ValueClause(Object object, Field field) {
         this.validateColumn(field);
-        this.object = object;
-        this.field = field;
+        this.value = extractValue(object, field);
     }
 
     public String getValue() {
+        return this.value;
+    }
+
+    private String extractValue(Object object, Field field) {
         if (field.isAnnotationPresent(Id.class)) {
-            return getPKValue();
+            return extractValueByPK(object, field);
         }
-        return getValueByField();
+        return extractValueByField(object, field);
     }
 
-    private String getPKValue() {
-        GenerationType generationType = field.isAnnotationPresent(GeneratedValue.class)
-                ? field.getAnnotation(GeneratedValue.class).strategy()
-                : GenerationType.AUTO;
+    private String extractValueByPK(Object object, Field field) {
+        GenerationType generationType = GenerationType.of(object.getClass());
```

Comment:
```suggestion
        GenerationType generationType = GenerationType.of(field);
```
로직상 field 넘겨주어도 될 것 같네요 :) 이미 찾았는데 또 클래스의 모든 필드에 대해서 Id 컬럼을 찾는 탐색이 발생하는것 같아요.
다른 곳에서도 GenerationType of(Class<?> clazz) 메소드는 호출이 필요 없어 보입니다.